src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -465,6 +500,72 @@ private void addFeedbackToResult(Result result, TestResultsDTO report) {
         }
     }
 
+    private List<BuildLogEntry> parsePipelineLogs(Element logHtml) throws IllegalArgumentException {
+        final var buildLog = new LinkedList<BuildLogEntry>();
+        if (logHtml.childNodes().stream().noneMatch(child -> child.attr("class").contains("pipeline"))) {
+            throw new IllegalArgumentException("Log is not pipeline log");
+        }
+        for (Element elem : logHtml.children()) {
+            // Only pipeline-node-ID elements contain actual log entries
+            if (elem.attributes().get("class").contains("pipeline-node")) {
+                // At least one child must have a timestamp class
+                if (elem.childNodes().stream().anyMatch(child -> child.attr("class").contains("timestamp"))) {
```

Comment:
Codacy is complaining here: `These nested if statements could be combined`

Reply:
Fwiw, Imho this would be more readable as
```
for (Element elem : logHtml.children()) {
    // Only pipeline-node-ID elements containing actual log entries with at least one child with a timestamp class
    if (!elem.attributes().get("class").contains("pipeline-node") || !elem.childNodes().stream().anyMatch(child -> child.attr("class").contains("timestamp"))) {
        continue;
    }

    Iterator<Node> nodeIterator = elem.childNodes().iterator();
    [...]
}
```
but that is personal preference.

Reply:
Fwiw, this was not resolved.(?)

Reply:
Thanks for the reminder, I actually prefer the current version if that is also fine for you.